tests/test_cli_configure.py

```diff
@@ -0,0 +1,153 @@
+    assert result.exit_code == 0
+
+    config = NbAutoexportConfig.parse_file(
+        path=tmp_path / SAVE_PROGRESS_INDICATOR_FILE, content_type="application/json"
+    )
+
+    expected_config = NbAutoexportConfig()
+    assert config == expected_config
+
+
+def test_configure_specified(tmp_path):
+    export_formats = ["script", "html"]
+    organize_by = "extension"
+    clean = True
+    assert export_formats != DEFAULT_EXPORT_FORMATS
+    assert organize_by != DEFAULT_ORGANIZE_BY
+    assert clean != DEFAULT_CLEAN
+
+    cmd_list = ["configure", str(tmp_path)]
+    for fmt in export_formats:
+        cmd_list.extend(["-f", fmt])
+    cmd_list.extend(["-b", organize_by])
+    if clean:
+        cmd_list.append("--clean")
+
+    result = CliRunner().invoke(app, cmd_list)
+    assert result.exit_code == 0
+
+    config = NbAutoexportConfig.parse_file(
+        path=tmp_path / SAVE_PROGRESS_INDICATOR_FILE, content_type="application/json"
+    )
+
+    expected_config = NbAutoexportConfig(
+        export_formats=export_formats, organize_by=organize_by, clean=clean
+    )
+    assert config == expected_config
+
+
+def test_invalid_export_format():
+    runner = CliRunner()
+    result = runner.invoke(app, ["configure", "-f", "invalid-output-format"])
+    assert result.exit_code == 2
+    assert (
+        "Error: Invalid value for '--export-format' / '-f': invalid choice: invalid-output-format"
+        in result.output
+    )
+
+
+def test_invalid_organize_by():
+    runner = CliRunner()
+    result = runner.invoke(app, ["configure", "-b", "invalid-organize-by"])
+    assert result.exit_code == 2
+    assert (
+        "Invalid value for '--organize-by' / '-b': invalid choice: invalid-organize-by"
+        in result.output
+    )
+
+
+def test_refuse_overwrite(tmp_path):
+    (tmp_path / ".nbautoexport").touch()
+    runner = CliRunner()
+    result = runner.invoke(app, ["configure", str(tmp_path)])
+    assert result.exit_code == 1
+    assert "Detected existing autoexport configuration at" in result.output
+
+
+def test_force_overwrite(tmp_path):
+    (tmp_path / ".nbautoexport").touch()
+    runner = CliRunner()
+    result = runner.invoke(
+        app, ["configure", str(tmp_path), "-o", "-f", "script", "-f", "html", "-b", "notebook"]
+    )
+    assert result.exit_code == 0
+    with (tmp_path / ".nbautoexport").open("r") as fp:
+        config = json.load(fp)
+
+    expected_config = NbAutoexportConfig(export_formats=["script", "html"], organize_by="notebook")
+    assert config == expected_config
+
+
+def test_install_no_jupyter_config_warning(tmp_path, monkeypatch):
+    def mock_jupyter_config_dir():
+        return str(tmp_path)
+
+    monkeypatch.setattr(jupyter_config, "jupyter_config_dir", mock_jupyter_config_dir)
+
+    result = CliRunner().invoke(app, ["configure", str(tmp_path)])
+    assert result.exit_code == 0
+    assert "Warning: nbautoexport is not properly installed with Jupyter." in result.output
```

Comment:
This test and `test_install_no_initialize_warning` are failing. This test should try to read a non-existent config from the `tmp_path` but instead it appears to be reading from my actual Jupyter config file (which contains a pre-versioning version of the post-save hook). I suppose the `mock_jupyter_config_dir` does not mock the `jupyter_config_dir()` call that happens _within_ the CLI call.

```
_____________________________________________ test_install_no_jupyter_config_warning _____________________________________________
                                                                                                                                  
tmp_path = PosixPath('/tmp/pytest-of-robert/pytest-0/test_install_no_jupyter_config0')                                            
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f678e302550>                                                          
                                
    def test_install_no_jupyter_config_warning(tmp_path, monkeypatch):
        def mock_jupyter_config_dir():     
            return str(tmp_path)                                                                                                  
                                                                                                                                  
        monkeypatch.setattr(jupyter_config, "jupyter_config_dir", mock_jupyter_config_dir)
                  
        result = CliRunner().invoke(app, ["configure", str(tmp_path)])                                                           
        assert result.exit_code == 0
>       assert "Warning: nbautoexport is not properly installed with Jupyter." in result.output
E       assert 'Warning: nbautoexport is not properly installed with Jupyter.' in "Warning: nbautoexport initialize is an older ve
rsion. Please run 'install' command to update.\n"
E        +  where "Warning: nbautoexport initialize is an older version. Please run 'install' command to update.\n" = <Result okay
>.output                                                                                                                          

tests/test_cli_configure.py:108: AssertionError                                                                                   
_______________________________________________ test_install_no_initialize_warning _______________________________________________
                                
tmp_path = PosixPath('/tmp/pytest-of-robert/pytest-0/test_install_no_initialize_war0')
monkeypatch = <_pytest.monkeypatch.MonkeyPatch object at 0x7f678e2f6e80>                  
                                
    def test_install_no_initialize_warning(tmp_path, monkeypatch):    
        def mock_jupyter_config_dir():
            return str(tmp_path)                                                                                                  
                                                                                                                                          monkeypatch.setattr(jupyter_config, "jupyter_config_dir", mock_jupyter_config_dir)
                                                                                                                                          (tmp_path / "jupyter_notebook_config.py").touch()
     
        result = CliRunner().invoke(app, ["configure", str(tmp_path)])
        assert result.exit_code == 0                                                                                              
>       assert "Warning: nbautoexport is not properly installed with Jupyter." in result.output
E       assert 'Warning: nbautoexport is not properly installed with Jupyter.' in "Warning: nbautoexport initialize is an older ve
rsion. Please run 'install' command to update.\n"                                                                                 
E        +  where "Warning: nbautoexport initialize is an older version. Please run 'install' command to update.\n" = <Result okay
>.output      
                                                                                                                                  
tests/test_cli_configure.py:121: AssertionError
```

Reply:
I'm a little surprised this happened given that the other tests seemed to pass, but I'm trying a different approach which I am confident should work.